File: Selenium/Activities/Activity11.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions
  
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Declare actions variable
    wait = WebDriverWait(driver, 20)
   
    # Navigate to the URL
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    print("Page title is: ", driver.title)

    checkbox = driver.find_element(By.CSS_SELECTOR,"input.willDisappear")
    

    print("Checkbox is selected",checkbox.is_selected())
    
    checkbox.click()

    print("Checkbox is selected",checkbox.is_selected())
      
    driver.close()

File: Selenium/Activities/Activity11.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;

import java.time.Duration;

public class Activity11 {
    public static void main(String[] args) {
        WebDriver driver = new FirefoxDriver();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Page title is: "+driver.getTitle());

        WebElement checkbox = driver.findElement(By.cssSelector("input.willDisappear"));

        checkbox.click();

        System.out.println("Checkbox is selected"+checkbox.isSelected());

        checkbox.click();

        System.out.println("Checkbox is selected"+checkbox.isSelected());

        driver.close();
    }
}
